ext/gvltools/instrumentation.c

```diff
@@ -67,16 +67,32 @@ static VALUE gt_disable_metric(VALUE module, VALUE metric) {
 }
 
 // GVLTools::LocalTimer and GVLTools::GlobalTimer
-static rb_atomic_t global_timer_total = 0;
+static rb_atomic_t global_timer_lock = 0;
+static uint64_t global_timer_total = 0;
 static THREAD_LOCAL_SPECIFIER uint64_t local_timer_total = 0;
 static THREAD_LOCAL_SPECIFIER struct timespec timer_ready_at = {0};
 
 static VALUE global_timer_monotonic_time(VALUE module) {
-    return UINT2NUM(global_timer_total);
+    uint64_t total;
+    while (1) {
+        if (0 == RUBY_ATOMIC_CAS(global_timer_lock, 0, 1)) {
```

Comment:
>  I have no idea how to do that. My C is quite rusty.

If we scope it to only supporting clang and GCC, it's not particularly hard, we can just rip Ruby's atomic.h and switch `unsigned int` for `uint64_t`. 

We could even just require `C11`, use `_Atomic` and call it a day. I don't think we need maximum portability.

Feel free to explore this. Otherwise I may find some time to look at it next week or so.